### PR TITLE
[cherry-pick][branch-2.3][enhancement]parquet format to support coalesce reads

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

cherry-pick from https://github.com/StarRocks/starrocks/pull/7478

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/formats/parquet/page_reader.cpp [hunk 1]
```diff
@@ -31,25 +29,24 @@ Status PageReader::next_header() {
 
     uint32_t header_length = 0;
     size_t nbytes = kHeaderBufSize;
-
+    size_t remaining = _finish_offset - _offset;
     do {
-        RETURN_IF_ERROR(_stream.get_bytes(&page_buf, &nbytes, true));
+        nbytes = std::min(nbytes, remaining);
+        RETURN_IF_ERROR(_stream->get_bytes(&page_buf, _offset, &nbytes));
```
Comment (DorianZheng) [inline be/src/formats/parquet/page_reader.cpp:35]: without the peek, the data read size might double, you should merge @GavinMar bug fix

File: be/src/util/buffered_stream.h [hunk 2]
```diff
@@ -64,4 +72,41 @@ class BufferedInputStream {
...
+class SharedBufferedInputStream : public IBufferedInputStream {
+public:
+    struct IORange {
+        int64_t offset;
+        int64_t size;
+        bool operator<(const IORange& x) const { return offset < x.offset; }
+    };
+    struct CoalesceOptions {
+        static constexpr int64_t MB = 1024 * 1024;
+        int64_t max_read_size = 16 * MB;
```
Comment (DorianZheng) [inline be/src/util/buffered_stream.h:84]: merge this pr https://github.com/StarRocks/starrocks/commit/5bf0be63dda73e119decf0cec72b36a669fe54a1

Comment (dirtysalt) [reply be/src/util/buffered_stream.h:84]: OK.

**Comment by dirtysalt** [comment]
Waiting for this PR passed. And I'll merge it later. https://github.com/StarRocks/starrocks/pull/9682

**Review by imay — APPROVED** [review]

**Pull request merged**